Framework/DataHandling/src/LoadSESANS.cpp

```diff
@@ -0,0 +1,419 @@
+    // attributes.end()){
+    // if (attributes.find(attr) == attributes.end()) {
+    if (!attributes.count(attr)) {
+      std::string err = "Failed to supply parameter: \"" + attr + "\"";
+      g_log.error(err);
+      throw std::runtime_error(err);
+    }
+  }
+}
+
+/** Create a new workspace with the columns read from the file
+ * @param columns Data columns from input file
+ * @return A workspace with the corresponding data
+ */
+API::MatrixWorkspace_sptr LoadSESANS::makeWorkspace(ColumnMap columns) {
+  size_t histogramLength = columns["SpinEchoLength"].size();
+  API::MatrixWorkspace_sptr newWorkspace =
+      API::WorkspaceFactory::Instance().create(
+          "Workspace2D", 1, histogramLength, histogramLength);
+
+  auto xValues = columns["Wavelength"];
+  auto yValues =
+      calculateYValues(columns["Depolarisation"], columns["Wavelength"]);
+  auto eValues = calculateEValues(columns["Depolarisation_error"], yValues,
+                                  columns["Wavelength"]);
+
+  auto &dataX = newWorkspace->mutableX(0);
+  auto &dataY = newWorkspace->mutableY(0);
+  auto &dataE = newWorkspace->mutableE(0);
+
+  for (size_t i = 0; i < histogramLength; i++) {
+    dataX[i] = xValues[i];
+    dataY[i] = yValues[i];
+    dataE[i] = eValues[i];
+  }
+
+  return newWorkspace;
+}
+
+/** Is a character whitespace (here considered space or tab)?
+ * @param c The character
+ * @return Whether it is whitespace
+ */
+bool LoadSESANS::space(const char &c) { return c == ' ' || c == '\t'; }
+
+/** Is a character not whitespace (here considered space or tab)?
+* @param c The character
+* @return Whether it is not whitespace
+*/
+bool LoadSESANS::notSpace(const char &c) { return !space(c); }
+
+/** Split a string on spaces
+ * @param str The string to split
+ * @return Vector of string segments
+ */
+std::vector<std::string> LoadSESANS::split(const std::string &str) {
+  std::vector<std::string> result;
+
+  auto i = str.begin();
+  i = find_if(i, str.end(), &notSpace);
+
+  auto j = find_if(i, str.end(), &space);
+
+  while (i != str.end()) {
+    result.push_back(std::string(i, j));
+    i = find_if(j, str.end(), &notSpace);
+    j = find_if(i, str.end(), &space);
+  }
+
+  return result;
+}
+
+/** Is a string all whitespace?
+ * @param str The string to test
+ * @return Whether every character is whitespace
+ */
+bool LoadSESANS::allSpaces(const std::string &str) {
+  return std::all_of(str.begin(), str.end(), space);
+}
+
+/** Repeat a string n times, delimited by another string. eg. repeatAndJoin("a",
+ * "b", 3) == "ababa"
+ * @param str The string to repeat
+ * @param delim The delimiter
+ * @param n The number of times to repeat
+ * @return The repeated string
+*/
+std::string LoadSESANS::repeatAndJoin(const std::string &str,
+                                      const std::string &delim, const int &n) {
+  std::string result = "";
+  for (int i = 0; i < n - 1; i++) {
+    result += str + delim;
+  }
+  return result + str;
```

Comment:
It's the other way round - so the regex goes `<number><space><number><space><number>`. There can be spaces on the end but it's not required